docs/make.jl

```diff
@@ -28,6 +28,7 @@ const PAGES = [
             "users_guide/QuantumObject/QuantumObject.md",
             "users_guide/QuantumObject/QuantumObject_functions.md",
         ],
+        "The Importance of Type-Stability" => "users_guide/type_stability.md",
```

Comment:
Maybe put this page under "Basic operations on quantum objects" ?

Reply:
not necessary, I just thought it might be reasonable cause the type of `dims` is part of `QuantumObject`

Reply:
My plan was also to discuss about type instabilities in general. For example, we have cases where it is recommended to use `Val(argument` instead of `argument` in a function. So I will make some practical examples which don't necessary involve `QuantumObject`

Reply:
So I decide to make a dedicated page, also for emphasize how important is type-stability for those who want maximum performances.

Reply:
okay